configs/components/puppet.rb

```diff
@@ -1,6 +1,10 @@
 component "puppet" do |pkg, settings, platform|
   pkg.load_from_json("configs/components/puppet.json")
 
+  if platform.is_windows?
+    pkg.add_source("file://resources/files/windows/environment.bat", sum: "c698da37e559935e904f8690dd5c26fa")
```

Comment:
what about adding environment.bat to puppet-agent itself, e.g. https://github.com/puppetlabs/puppet-agent/tree/master/resources/windows/msi/environment.bat